lib/data/managers/firestore_manager.dart

```diff
@@ -113,13 +97,59 @@ class FireStoreManager {
     }
     return articlesEntitiesWithImagePath;
   }
+
   Future<String> getImageDownloadURL(DocumentSnapshot imageDocument) async {
-    final sizePath = imageDocument.data["sizes"].first["path"];
-    final imageFileNamePath = imageDocument.data["file"];
+    final sizePath = imageDocument.data[imageSizes].first[imagePath];
+    final imageFileNamePath = imageDocument.data[imageFile];
     final flamelinkPath =
         IMAGE_BASE_PATH + '/' + sizePath + '/' + imageFileNamePath;
     final storageReference =
         FirebaseStorage.instance.ref().child(flamelinkPath);
     return await storageReference.getDownloadURL();
   }
+
+  Future<ArticlesDirectoryEntity> getArticlesDirectory() async {
+    ArticlesDirectoryEntity articlesDirectory;
+
+    // Filters defined by product definition. - ARTICLES DIRECTORY
+    var query = Firestore.instance
+        .collection(FLAME_LINK_CONTENT)
+        .where(FLAME_LINK_SCHEMA, isEqualTo: Schema.ARTICLE_DIRECTORY)
+        .where(FLAME_LINK_SCHEMA_TYPE, isEqualTo: SchemaType.SINGLE)
+        .where(FLAME_LINK_ENVIROMENT,
+            isEqualTo: FirestoreEnvironment.PRODUCTION)
+        .where(FLAME_LINK_LOCALE, isEqualTo: Locale.EN_US);
+
+    await query.getDocuments().then((snapshot) {
+      articlesDirectory = snapshot.documents.map((articlesDirectoryDocument) {
+        return ArticlesDirectoryEntity.featuredArticlesFromDocument(
+            articlesDirectoryDocument);
+      }).first;
+    });
+    return articlesDirectory;
+  }
+
+  Future<ArticlesDirectoryEntity> getFeaturedArticles(
+      ArticlesDirectoryEntity articlesDirectory) async {
+    ArticlesDirectoryEntity articlesDirectoryWithFeaturedArticles;
+
+    if (articlesDirectory.articlesPathReference != null) {
+      for (var articlePathReference
+          in articlesDirectory.articlesPathReference) {
+        await Firestore.instance
+            .document(articlePathReference)
+            .get()
+            .then((featuredArticlesSnapshot) async {
+          if (featuredArticlesSnapshot.data != null &&
+              featuredArticlesSnapshot.data[documentFieldStatus] ==
+                  DataStatus.PUBLISHED) {
+            articlesDirectory.addArticle(
+                ArticleEntity.articleFromDocument(featuredArticlesSnapshot));
+          }
+        });
+      }
+    }
+    articlesDirectoryWithFeaturedArticles = articlesDirectory;
+    return articlesDirectoryWithFeaturedArticles;
```

Comment:
If changes are done based on previous comment, then these last two lines won't unnecessarily create extra references to the same object saving some amount of memory.